ldap/servers/slapd/conntable.c

```diff
@@ -48,69 +48,61 @@
  * under the connection table lock. This should move the allocation algorithm from O(n) worst case
  * to O(1) worst case as we always recieve and empty slot *or* ct full. It also reduces lock/atomic
  * contention on the CPU to improve things.
+ * Lastly a change was done: Instead of having a sliding windows that tend to get a never used
+ * slot for each new connection, it nows reuse last freed one. That has several benefits:
+ *  - Fix a bug because the last free list slaots may not be alloced.
+ *  - Avoid to grow the memory footprint when there is no much load
+ *  - Simplify the code (as only a single is needed. )
  *
- * The freelist is a ringbuffer of pointers to the connection table. On a small scale it looks like:
+ * The freelist is a ringbuffer of pointers to the connection table.
+ * It is NULL terminated. On a small scale it looks like:
  *
  *  |--------------------------------------------|
- *  | slot 1 | slot 2 | slot 3 | slot 4 | slot 5 |
- *  | _ ptr  | _ ptr  | _ ptr  | _ ptr  | _ ptr  |
+ *  | slot 0 | slot 1 | slot 2 | slot 3 | slot 4 |
+ *  | _ ptr  | _ ptr  | _ ptr  | _ ptr  | NULL   |
  *  |--------------------------------------------|
- *     ^  ^- conn_next
- *     |
- *     \-- conn_free
+ *        ^- conn_next
  *
  * As we allocate, we shift conn_next through the list, yielding the ptr that was stored (and
  * setting it to NULL as we proceed)
  *
  *  |--------------------------------------------|
- *  | slot 1 | slot 2 | slot 3 | slot 4 | slot 5 |
- *  | _NULL  | _NULL  | _ ptr  | _ ptr  | _ ptr  |
+ *  | slot 0 | slot 1 | slot 2 | slot 3 | slot 4 |
+ *  | _NULL  | _NULL  | _ ptr  | _ ptr  | NULL   |
  *  |--------------------------------------------|
- *     ^                  ^- conn_next
- *     |
- *     \-- conn_free
+ *                        ^- conn_next
  *
  * When a connection is "freed" we return it to conn_free, which is then also slid up.
```

Comment:
conn_free removed, so comment needs update.

Reply:
Also mention that conn_next is moving back to the left here for the freed conn ptr to occup the slot to the left.